src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -217,7 +219,7 @@ private static void finishInitialize(ConfigurationAnnotation result, IConfigurat
 //    for (Class<?> cls : classes) {
         while (null != cls) {
           boolean hasClassAnnotation = isAnnotationPresent(annotationFinder, cls, annotationClass);
-          Method[] methods = cls.getDeclaredMethods();
+          Method[] methods = getDeclaredMethods(cls);
```

Comment:
I checked and as I understand, on Java8, `cls.getDeclaredMethods()` will return default methods too.
If I'm true, could you add a comment which says what we will have to do when java7 will be dropped?